### PR TITLE
saveSubmissionsInSQLite

### PR DESCRIPTION
**Pull request by jeffreykat**
Accessing data on View Submissions Activity is erroring

### CODE REVIEW

**Comment by Bradford-Wong** [comment]
I looked at it for a couple of hours today. I don't know if you made any more progress, but here are my findings:

1. I'm pretty sure that the database is storing information correctly. I was able to extract the database and look at the contents of it using a third party app. This made me think that the problem was with reading the data.
2. In getSubmissionTitles() (from myDBHandler.java), I replaced " titles[i] = cursor.getString(cursor.getPosition());" (line 184) with "titles[i] = cursor.getString(cursor.getColumnIndex(Submission.COLUMN_TITLE));"
3. I temporarily commented out "subsDates = dbHandler.getSubmissionDates();" in ViewSubsActivity.java
4. When setting the adapter (again in ViewSubsActivity.java), I used subsTitles for all the arrays instead of using subsDates and subsCaseID (although using subsCaseID will probably work)

Doing all of that made the app work. I haven't looked at subsDates yet, but the also app crashes whenever "subsDates = dbHandler.getSubmissionDates();" is executed. I suspect there is probably a similar issue for subsDates and a fix similar to what I did in step 2. Hope that helps!

File: app/src/main/java/com/myvetpath/myvetpath/ViewDraftsActivity.java [hunk 1]
```diff
@@ -1,16 +1,114 @@
...
+        AlertDialog.Builder dialog = new AlertDialog.Builder(ViewDraftsActivity.this);
+        dialog.setCancelable(true);
+        String title = getString(R.string.action_delete_confirmation_prompt_first_part)
+                + drafts[selectedSubmissionPosition].getTitle()
+                + getString(R.string.action_delete_confirmation_second_part);
+        dialog.setTitle(title);
+        dialog.setPositiveButton(R.string.action_yes, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialogInterface, int i) {
+                Toast.makeText(ViewDraftsActivity.this, R.string.deleted_message,
```
Comment (Bradford-Wong) [inline app/src/main/java/com/myvetpath/myvetpath/ViewDraftsActivity.java]: We should probably say which submission the user just deleted.

File: app/src/main/java/com/myvetpath/myvetpath/ViewSubsActivity.java [hunk 2]
```diff
@@ -17,66 +19,46 @@
...
+        AlertDialog.Builder dialog = new AlertDialog.Builder(ViewSubsActivity.this);
+        dialog.setCancelable(true);
+        String title = getString(R.string.action_delete_confirmation_prompt_first_part)
+                + submissions[selectedSubmissionPosition].getTitle()
+                + getString(R.string.action_delete_confirmation_second_part);
+        dialog.setTitle(title);
+        dialog.setPositiveButton(R.string.action_yes, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialogInterface, int i) {
+                Toast.makeText(ViewSubsActivity.this, R.string.deleted_message,
```
Comment (Bradford-Wong) [inline app/src/main/java/com/myvetpath/myvetpath/ViewSubsActivity.java]: Would be useful if we said which submission was deleted

Comment (Bradford-Wong) [inline app/src/main/java/com/myvetpath/myvetpath/ViewSubsActivity.java]: Would be useful if we said which submission was deleted

File: app/src/main/java/com/myvetpath/myvetpath/SubDetailsActivity.java [hunk 3]
```diff
@@ -14,13 +14,18 @@
...
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_sub_details);
-        int position = getIntent().getIntExtra("pos", 0);
-        String title = getResources().getStringArray(R.array.subsTitles)[position];
+        myDBHandler = new MyDBHandler(this);
+
+        int caseId = getIntent().getIntExtra("caseID", 1);
```
Comment (Bradford-Wong) [inline app/src/main/java/com/myvetpath/myvetpath/SubDetailsActivity.java]: From the Design Document "Case ID: An integer number created by using the last two digits of the year and then a six digit runningtotal. It’s primary use is to allow a pathologist to easily recognizing what submission they are working on.May be dropped in the future when the Use of this app expands."

**Comment by jeffreykat** [comment]
I will fix CaseID based on meeting discussion. CaseID will be set to 0 and display String "Pending" until it is assigned by the server.

File: app/src/main/java/com/myvetpath/myvetpath/CreateSubActivity.java [hunk 4]
```diff
@@ -30,6 +65,47 @@ protected void onCreate(Bundle savedInstanceState) {
...
+
+        //initialize submission elements
+        title_et = findViewById(R.id.sub_title);
+        save_draft_button = findViewById(R.id.save_draft_btn);
+        save_draft_button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                hideSoftKeyboard();
+                long curDate = Calendar.getInstance().getTime().getTime();
+                newSub.setCaseID(NULL);
```
Comment (Bradford-Wong) [inline app/src/main/java/com/myvetpath/myvetpath/CreateSubActivity.java]: A comment explaining why we're always setting this to null may be useful for whoever has to change this later.

File: app/src/main/java/com/myvetpath/myvetpath/CreateSubActivity.java [hunk 5]
```diff
@@ -30,6 +65,47 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                newSub.setTitle(title_et.getText().toString());
+                newSub.setStatusFlag(0);
+                newSub.setDateOfCreation(curDate);
+                //Display confirmation Toast
+                String content = title_et.getText().toString() + " Saved";
+                Toast testToast = Toast.makeText(getApplicationContext(), content, Toast.LENGTH_LONG);
+                testToast.show();
+                dbHandler.addSubmission(newSub);
```
Comment (Bradford-Wong) [inline app/src/main/java/com/myvetpath/myvetpath/CreateSubActivity.java]: Later on we may also want to add a dialog box here to keep it consistent with the submit button.

File: app/src/main/java/com/myvetpath/myvetpath/MainActivity.java [hunk 6]
```diff
@@ -97,14 +104,9 @@ public void onClick(View view) {
...
 
-
-        view_subs_button = findViewById(R.id.viewSubsButton);
-        view_subs_button.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                startActivity(view_subs_activity);
-            }
-        });
+        if(dbHelper.getNumberOfDrafts() == 0){
```
Comment (Bradford-Wong) [inline app/src/main/java/com/myvetpath/myvetpath/MainActivity.java]: We may also want to do the same for the submit button later to keep it consistent. Next time, I would also appreciate a heads up on a change like this before I start testing.

**Comment by BJolly221** [comment]
Once the conflicts have been fixed, it should be fine to merge it.

**Pull request merged**